### PR TITLE
firefox-portable: Add scripts to set as default browser

### LINKED ISSUE #4090: set firefox as default browser
I just find Windows 10 does not support non-installed programs in **System - Default apps**. If you use Google Chrome and choose "set as default" in Chrome's setting, it will write itself into registry and you can find Chrome in **System - Default apps**. While if one installs Firefox using scoop and click on "set as default" in Firefox setting, one will be navigated into the **System - Default apps**, like . But Firefox does not write itself into registry and you can not find Firefox in **Choose an application** dialog.

I find one solution [here](https://www.winhelponline.com/blog/register-firefox-portable-with-default-programs-in-vista/). It works but it is not open source. Anyone has other ideas? I guess a post-installed script may help.

### PR DESCRIPTION
**Pull request by QingKongBaiYu**
fix #4090 .

### CODE REVIEW

File: bucket/firefox-portable.json [hunk 1]
```diff
@@ -21,6 +38,28 @@
...
     ],
+    "post_install": [
+        "if (Test-Path \"$dir\\firefox-install.reg\") {",
+        "  $fpath = \"$dir\".Replace('\\', '\\\\')",
+        "  $content = Get-Content \"$dir\\firefox-install.reg\"",
+        "  $content = $content.Replace('<path>', $fpath)",
+        "  if ($global) {",
+        "    $content = $content.Replace('HKEY_CURRENT_USER', 'HKEY_LOCAL_MACHINE')",
+        "  }",
+        "  $content | Set-Content -Path \"$dir\\vscode-install-context.reg\"",
```
Comment (Ash258) [inline bucket/firefox-portable.json:49]: nonsense

File: bucket/firefox-portable.json [hunk 2]
```diff
@@ -21,6 +38,28 @@
...
+        "  $content | Set-Content -Path \"$dir\\firefox-install.reg\"",
+        "}",
+        "if (Test-Path \"$dir\\firefox-uninstall.reg\") {",
+        "  $fpath = \"$dir\".Replace('\\', '\\\\')",
+        "  $content = Get-Content \"$dir\\firefox-uninstall.reg\"",
+        "  $content = $content.Replace('<path>', $fpath)",
+        "  if ($global) {",
+        "    $content = $content.Replace('HKEY_CURRENT_USER', 'HKEY_LOCAL_MACHINE')",
+        "  }",
+        "  $content | Set-Content -Path \"$dir\\vscode-uninstall-context.reg\"",
```
Comment (Ash258) [inline bucket/firefox-portable.json:59]: nonsense

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Comment by Ash258** [comment]
/unsubscribe

I do not think more scripts like this should be introduced in official buckets.

**Comment by QingKongBaiYu** [comment]
> /unsubscribe
> 
> I do not think more scripts like this should be introduced in official buckets.

sorry for the mistake!

**Pull request closed without merging**